ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/util/EpochProcessorUtil.java

```diff
@@ -345,13 +347,19 @@ private static UnsignedLong get_base_reward(BeaconState state, int index) {
       }
     }
 
+    Map<PendingAttestation, HashSet<Integer>> attesting_indices =
+        matching_source_attestations.stream()
```

Comment:
Maybe you could do this in parallel to speed this up even more?

Reply:
Honestly I would avoid 'micro-parallelizations' for now, except of evident cases like BLS signature check. The code inside the loop is not too heavy and the parallelization gain is questionable. 
First it could make perf measurements more dispersed. 
Second: event if we win something now it could be neglected if caches would be used inside the loop body when the cache would need to be accessed concurrently.
Third I would target more high level functions for parallelization if possible